var/spack/repos/builtin/packages/sw4/package.py

```diff
@@ -8,61 +8,72 @@
+    variant("hdf5", default=True, description="Build with HDF5")
+    variant("zfp", default=False, when="+hdf5", description="Build with ZFP")
+    variant("fftw", default=True, description="Build with FFTW")
+    variant("debug", default=False, description="Build with debugging symbols")
 
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
     depends_on("mpi")
     depends_on("blas")
     depends_on("lapack")
-    depends_on("proj", when="+proj")
-    depends_on("hdf5+mpi", when="+hdf5")
-    depends_on("fftw@3.0: +mpi", when="+fftw")
+    depends_on("python")
+    depends_on("proj@9:", when="+proj")
+    depends_on("hdf5@1.14: +mpi", when="+hdf5")
+    depends_on("py-h5py", when="+hdf5")
     depends_on("zfp", when="+zfp")
     depends_on("h5z-zfp@develop", when="+zfp")
-    depends_on("python")
-    depends_on("py-h5py")
+    depends_on("fftw@3: +mpi", when="+fftw")
     depends_on("llvm-openmp", when="%apple-clang +openmp")
 
-    def edit(self, spec, prefix):
-        os.environ["CXX"] = spec["mpi"].mpicxx
-        os.environ["FC"] = spec["mpi"].mpifc
-        os.environ["proj"] = "yes"
-        os.environ["openmp"] = "yes"
-        os.environ["hdf5"] = "yes"
-        os.environ["zfp"] = "yes"
-        os.environ["fftw"] = "yes"
-        os.environ["SW4ROOT"] = spec["proj"].prefix
-        os.environ["HDF5ROOT"] = spec["hdf5"].prefix
-        os.environ["H5ZROOT"] = spec["h5z-zfp"].prefix
-        os.environ["ZFPROOT"] = spec["zfp"].prefix
-        os.environ["FFTWHOME"] = spec["fftw"].prefix
+    def setup_build_environment(self, env):
+        env.set("CXX", self.spec["mpi"].mpicxx)
+        env.set("FC", self.spec["mpi"].mpifc)
+        # openmp is enabled by default
+        if self.spec.satisfies("~openmp"):
+            env.set("openmp", "no")
+        if self.spec.satisfies("+proj"):
+            env.set("proj", "yes")
+            env.set("SW4ROOT", self.spec["proj"].prefix)
+        if self.spec.satisfies("+hdf5"):
+            env.set("hdf5", "yes")
+            env.set("HDF5ROOT", self.spec["hdf5"].prefix)
+        if self.spec.satisfies("+zfp"):
+            env.set("zfp", "yes")
+            env.set("ZFPROOT", self.spec["zfp"].prefix)
+            env.set("H5ZROOT", self.spec["h5z-zfp"].prefix)
+        if self.spec.satisfies("+fftw"):
+            env.set("fftw", "yes")
+            env.set("FFTWHOME", self.spec["fftw"].prefix)
+        if self.spec.satisfies("+debug"):
+            env.set("debug", "yes")
         os.environ["EXTRA_LINK_FLAGS"] = "-lstdc++ -lm -ldl "
-        os.environ["EXTRA_LINK_FLAGS"] += spec["blas"].libs.ld_flags + " "
-        os.environ["EXTRA_LINK_FLAGS"] += spec["blas"].libs.ld_flags + " "
-
-        if "+openmp" in spec:
-            if spec.satisfies("%apple-clang"):
-                os.environ["EXTRA_LINK_FLAGS"] += spec["llvm-openmp"].libs.ld_flags + " "
-
+        os.environ["EXTRA_LINK_FLAGS"] += self.spec["blas"].libs.ld_flags + " "
```

Comment:
```suggestion
        env.append_flags("EXTRA_LINK_FLAGS", self.spec["blas"].libs.ld_flags)
```
And similar below.

This ensures that environment modifications are limited to the build of the package only, even if a single spack process install multiple other packages.

Reply:
I wasn't aware of the `append_flags` function. These lines have been updated.